### ISSUE TITLE
[Bug] video_demo crashes and produces an upside down output

### ISSUE THREAD

**Issue by IECCLES4** [open-mmlab/mmdetection#9584]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

master branch https://github.com/open-mmlab/mmdetection

### Environment

sys.platform: linux
Python: 3.8.13 (default, Oct 21 2022, 23:50:54) [GCC 11.2.0]
CUDA available: True
GPU 0: NVIDIA GeForce GTX 1070
CUDA_HOME: /home/dtl-admin/miniconda3/envs/mmdet
NVCC: Cuda compilation tools, release 11.6, V11.6.124
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
PyTorch: 1.11.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.5.2 (Git Hash a9302535553c73243c632ad3c4c80beec3d19a1e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.2
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.11.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.12.0
OpenCV: 4.6.0
MMCV: 1.5.3
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 11.3
MMDetection: 2.25.3+2b5504e

### Reproduces the problem - code sample

```
# Adding a base config file
_base_ = '../faster_rcnn/faster_rcnn_r101_caffe_fpn_1x_coco.py'

# We also need to change the num_classes in head to match the dataset's annotation
model = dict(
    roi_head=dict(
        bbox_head=dict(num_classes=1)))

# Modify dataset related settings
dataset_type = 'COCODataset'
classes = ('pantograph',)
data = dict(
    train=dict(
        img_prefix='configs/pantograph/train/',
        classes=classes,
        ann_file='configs/pantograph/train/G-gauge_blackpool-trams_trainsim.json'),
    val=dict(
        img_prefix='configs/pantograph/val/',
        classes=classes,
        ann_file='configs/pantograph/val/PantoVal.json'),
    test=dict(
        img_prefix='configs/pantograph/val/',
        classes=classes,
        ann_file='configs/pantograph/val/PantoVal.json'))

# We can use the pre-trained Mask RCNN model to obtain higher performance
load_from = 'checkpoints/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth'
```

### Reproduces the problem - command or script

```
python demo/video_demo.py configs/pantograph/GX030025.MP4 configs/pantograph/faster_rcnn_pantograph.py download_2023-01-06_09-40-45/2023-01-05_G-Gauge_TrainSim2_Checkpoint/latest.pth --out balckpool_tram_results.mp4
```

### Reproduces the problem - error message

```
load checkpoint from local path: download_2023-01-06_09-40-45/2023-01-05_G-Gauge_TrainSim2_Checkpoint/latest.pth
[                                                  ] 0/7488, elapsed: 0s, ETA:/home/dtl-admin/dev/repos/bitbucket/railsight/mmdetection-2.25.3/mmdet/datasets/utils.py:66: UserWarning: "ImageToTensor" pipeline is replaced by "DefaultFormatBundle" for batch inference. It is recommended to manually replace it in the test data pipeline in your config file.
  warnings.warn(
[                                                  ] 55/7488, 2.2 task/s, elapsed: 25s, ETA:  3375s
(mmdet) dtl-admin@white-tower:~/dev/repos/bitbucket/railsight/mmdetection-2.25.3$
```

**Comment by IECCLES4** [comment]
I think I should also mention that when trying to train now the loss goes to nan extremely quickly on epoch 1

**Comment by BIGWangYuDong** [comment]
> I think I should also mention that when trying to train now the loss goes to nan extremely quickly on epoch 1

https://github.com/open-mmlab/mmdetection/blob/master/docs/en/faq.md

**Comment by BIGWangYuDong** [comment]
As for the first question, does not meet this error before, we will have a check

**Comment by IECCLES4** [comment]
> > I think I should also mention that when trying to train now the loss goes to nan extremely quickly on epoch 1
> 
> https://github.com/open-mmlab/mmdetection/blob/master/docs/en/faq.md#training

I've looked at this document but it doesn't seem to say where I change some of the values is it in training.py?

**Comment by BIGWangYuDong** [comment]
- Reduce the learning rate: the learning rate might be too large due to some reasons, e.g., change of batch size. You can rescale them to the value that could stably train the model.
- Extend the warmup iterations: some models are sensitive to the learning rate at the start of the training. You can extend the warmup iterations, e.g., change the warmup_iters from 500 to 1000 or 2000. **(change the `warmup_iters` in lr_config)**
- Add gradient clipping: some models requires gradient clipping to stabilize the training process. The default of grad_clip is None, you can add gradient clippint to avoid gradients that are too large, i.e., set optimizer_config=dict(_delete_=True, grad_clip=dict(max_norm=35, norm_type=2)) in your config file. If your config does not inherits from any basic config that contains optimizer_config=dict(grad_clip=None), you can simply add optimizer_config=dict(grad_clip=dict(max_norm=35, norm_type=2)).  **(change `optimizer_config`)**

**Comment by IECCLES4** [comment]
> * Reduce the learning rate: the learning rate might be too large due to some reasons, e.g., change of batch size. You can rescale them to the value that could stably train the model.
> 
>     * Extend the warmup iterations: some models are sensitive to the learning rate at the start of the training. You can extend the warmup iterations, e.g., change the warmup_iters from 500 to 1000 or 2000. **(change the `warmup_iters` in lr_config)**
> 
>     * Add gradient clipping: some models requires gradient clipping to stabilize the training process. The default of grad_clip is None, you can add gradient clippint to avoid gradients that are too large, i.e., set optimizer_config=dict(_delete_=True, grad_clip=dict(max_norm=35, norm_type=2)) in your config file. If your config does not inherits from any basic config that contains optimizer_config=dict(grad_clip=None), you can simply add optimizer_config=dict(grad_clip=dict(max_norm=35, norm_type=2)).  **(change `optimizer_config`)**
> 
> 
> This is suggestions, which you can have a try

The error was my PC isn't up for the job anymore however, it doesn't have an error so took me longer to figure out. Many thanks for your help.
